Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m

```diff
@@ -351,31 +388,55 @@ - (google_crashlytics_Session_Device)protoSessionDevice {
   return device;
 }
 
+- (NSUInteger)numberOfEvents {
+  NSUInteger sum = [self hasCrashed] ? 1 : 0;
+  sum += self.errors.count;
+  return sum;
+}
+
 - (google_crashlytics_Session_Event *)protoEvents {
-  google_crashlytics_Session_Event *events = malloc(sizeof(google_crashlytics_Session_Event) * 1);
+  // TODO: Add custom exceptions (when supported)
+  NSUInteger numberOfEvents = [self numberOfEvents];
 
-  // TODO:
-  // 1. Verify assumption that only one crash event per report
-  // 2. Handle reports with 0 crashes
-  // 3. Add events for non-fatal errors
-  events[0] = [self protoCrashEvent];
+  // Add recorded error events
+  google_crashlytics_Session_Event *events =
+      malloc(sizeof(google_crashlytics_Session_Event) * numberOfEvents);
+
+  for (NSUInteger i = 0; i < self.errors.count; i++) {
+    events[i] = [self protoEventForError:self.errors[i]];
+  }
+
+  // Add crash event
+  if (numberOfEvents > 0) {
+    events[numberOfEvents - 1] = [self protoEventForCrash];
+  }
```

Comment:
If we have 4 non-fatals, but no crash, I think this will still try to put a crash into the report?

Reply:
Nice catch, this was an artifact of some manual testing that I was doing. I'll change this back to what I had:
`if ([self hasCrashed)`